core/src/price_estimation/price_source.rs

```diff
@@ -36,6 +36,48 @@ impl PriceSource for NoopPriceSource {
     }
 }
 
+/// Wait for the price source to have a price for this token.
+/// Useful for ThreadedPriceSource to ensure that it has updated at least once.
+pub async fn wait_for_price(price_source: &dyn PriceSource, token_id: TokenId) {
```

Comment:
Maybe this should be implemented as an `initialize` method part of the `PriceSource` trait with a default no-op implementation (kind of how we do it with the `OrderbookReading`). This way the `ThreadedPriceSource` can use something like a `oneshot` channel to properly notify initialization instead of polling for token 0's price, which seems a bit incorrect to me, since its not part of the `PriceSource` contract that a price for token 0 *must* be provided. In fact, since token 0 has a known price, I wouldn't be surprised if a `PriceSource` never provided a price estimate for it.